Feature branch: interactive form

bini-i opened the pull request:
#### The following changes are introduced in this branch
- Connect BooksForm component to **redux** store
- made BooksForm to handle it's on state using **useState**
- Add **handleChange** method to BooksForm component that updates the states when a user
  interacts with BooksForm component
- Add **handleSubmit** method to BooksForm component that adds a new book to the redux store
- Connect Book component to the **redux** store
- Add **handleRemoveBook** method to Book component that removes the desired book

Commit:
Update BooksForm.js

- connect BooksForm component to redux store and passed only
  dispatch as props
- made BooksForm to handle its on state for (title, category and
  bookTitleError) using useState
- add handleChange method that update the states when user
  interacts with BooksForm component
- add handleSubmit method that adds new book to the redux store
  and intialize component states back (also handles book name
  input validation)

Commit:
Update Book.js

- connect the component to redux store using only dispatch as props
- add a remove book button to each row of books
- add handleRemoveBook method that handles removing a book
  from the redux store

Commit:
Update src/reducer/books.js

- update REMOVE_BOOK switch case to remove the desired book
  by filtering all the books except to be removed book

rindrajosia commented on src/components/BooksForm.js:
```diff
@@ -1,17 +1,55 @@
-        <form>
-            <input type="text" id="title"/>
-            <select>
+        <form style={{display: 'flex', justifyContent: 'center'}}>
```
- [ ] [REQUIRED] No inline styling: Make sure that basic CSS bad practices are not used [(article for support](https://speckyboy.com/good-bad-css-practices) ) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/section/css_bad_practices_requirement.png) 
from Pathwright)

- [x] [REQUIRED] Please attaches handleSubmit to `form's onSubmit` event [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/HandleSubmit.png)

rindrajosia commented on src/components/BooksList.js:
```diff
@@ -7,6 +7,7 @@ const BooksList = ({books}) => {
         const booksRow = books.map((book)=>
```
- [x] [REQUIRED] it should implement `handleBookRemove` [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/ImplementsRemoveBook.png)
- [x] [REQUIRED] It should connect the `REMOVE_BOOK` action.
- [ ] [REQUIRED] It should passe `handleRemoveBook` to Book component [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/ImplementsRemoveBook.png)

rindrajosia commented on src/components/Book.js:
```diff
@@ -1,19 +1,32 @@
+                <td>{title}</td>
+                <td>{category}</td>
+                <td>
+                    <button onClick={handleRemoveBook} name='removeBook' type="button" value='Remove book'>
```
- [x] [REQUIRED] Instead of attaching `handleRemoveBook` that you have implemented in this component, you should attach passed `handleRemoveBook` from `BookList` component to on click event of `Remove book` button.

rindrajosia commented on src/components/Book.js:
```diff
@@ -1,19 +1,32 @@
+    }
 
-export default Book;
+export default connect(null, mapDispatchToProps)(Book);
```
- [x] [REQUIRED] This component should stay a presentational component

rindrajosia reviewed (changes requested):
Hi, 

Good job so far!
There are some issues to revise before continuing to the next project but hang in there and read the comments below with care. You can do it! :muscle:


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

Commit:
Update BooksForm.js

- update the form to call handleSubmit onSubmit event

Commit:
Update BookList.js

- add handleRemoveBook event handler to handle the action
  of removing the specified book
- pass handleRemoveBook as a prop to Book component

Commit:
Update Book.js

- remove handleRemoveBook event handler and placed it
  inside BooksList component

Commit:
Refactor BooksForm.js and add styles.css

- removed all inline styles and reference them from styles.css

afanimartin commented on src/components/BooksForm.js:
```diff
@@ -34,16 +34,16 @@ const BooksForm = ({addNewBook}) => {
 
     return (
-        <form style={{display: 'flex', justifyContent: 'center'}}>
+        <form onSubmit={handleSubmit} style={{display: 'flex', justifyContent: 'center'}}>
```
Please remove the inline styling as pointed out by the previous reviewer. This is a bad CSS practice.

bini-i replied:
Thanks for the review
I think the changes are already made,

bini-i replied:
I think the changes are already made,
Thanks for the review

afanimartin reviewed (changes requested):
Hi @bini-i, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

### Martin Afani
#### Slack: whiz25

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

taiwo2 reviewed (approved):
# STATUS: APPROVED 🎉 
@bini-i well done and thanks for implementing the suggested changes, its time to merge
cheer 🥂 🥂
